src/compiler/checker.ts

```diff
@@ -21811,7 +21813,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             // No array like or unmatched property error - just issue top level error (errorInfo = undefined)
         }
 
-        function propertiesRelatedTo(source: Type, target: Type, reportErrors: boolean, excludedProperties: Set<__String> | undefined, intersectionState: IntersectionState): Ternary {
+        function propertiesRelatedTo(source: Type, target: Type, reportErrors: boolean, excludedProperties: Set<__String> | undefined, optionalsOnly: boolean, intersectionState: IntersectionState): Ternary {

```

Comment:
Rather than add a new boolean parameter to this function that we have to pass everywhere, could we not use the existing `excludedProperties` parameter to exclude all the non-optional properties at the single callsite we wanted to?

Reply:
Yeah, I considered that, but didn't want the overhead of constructing a new set, and the function is only called in three places.